src/ast/mod.rs

```diff
@@ -2844,6 +2847,41 @@ impl fmt::Display for ListAggOnOverflow {
     }
 }
 
+/// An `ARRAY_AGG` invocation `ARRAY_AGG( [ DISTINCT ] <expr> [ORDER BY <expr>] [LIMIT <n>] )
+/// [ WITHIN GROUP (ORDER BY <within_group>  ]`
+/// ORDERY BY position is defined differently for BigQuery, Postgres and Snowflake
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+pub struct ArrayAgg {
+    pub distinct: bool,
+    pub expr: Box<Expr>,
+    pub order_by: Option<Box<OrderByExpr>>,
+    pub limit: Option<Box<Expr>>,
```

Comment:
@SuperBo any reason why you did the implementation for all dialects the same time?

You didn't complete none, but added parts of different ones. Got me a little confused

Reply:
I changed this. You can review again